opencti-platform/opencti-front/src/private/components/data/stream/StreamCollectionEdition.jsx

```diff
@@ -76,56 +71,43 @@ const streamCollectionValidation = (t) => Yup.object().shape({
   name: Yup.string().required(t('This field is required')),
   description: Yup.string().nullable(),
   stream_public: Yup.bool().nullable(),
+  authorized_members: Yup.array().nullable(),
 });
 
-const groupMutationRelationAdd = graphql`
-  mutation StreamCollectionEditionGroupAddMutation($id: ID!, $groupId: ID!) {
-    streamCollectionEdit(id: $id) {
-      addGroup(id: $groupId) {
-        ...StreamCollectionEdition_streamCollection
-      }
-    }
-  }
-`;
-
-const groupMutationRelationDelete = graphql`
-  mutation StreamCollectionEditionGroupDeleteMutation($id: ID!, $groupId: ID!) {
-    streamCollectionEdit(id: $id) {
-      deleteGroup(id: $groupId) {
-        ...StreamCollectionEdition_streamCollection
-      }
-    }
-  }
-`;
-
-const StreamCollectionEditionContainer = (props) => {
-  const { t, classes, handleClose, streamCollection } = props;
-  const groups = pipe(
-    pathOr([], ['groups']),
-    map((n) => ({
-      label: n.name,
-      value: n.id,
-    })),
-  )(streamCollection);
-  const initialValues = { ...streamCollection };
-  initialValues.groups = groups;
+const StreamCollectionEditionContainer = ({ handleClose, streamCollection }) => {
+  const classes = useStyles();
+  const { t } = useFormatter();
+  const authorizedMembers = streamCollection.authorized_members?.map(({ id, name }) => ({ value: id, label: name })) ?? [];
+  const initialValues = { ...streamCollection, authorized_members: authorizedMembers };
   const [filters, setFilters] = useState(
-    JSON.parse(props.streamCollection.filters),
+    JSON.parse(streamCollection.filters),
   );
   const handleSubmitField = (name, value) => {
-    streamCollectionValidation(props.t)
+    streamCollectionValidation(t)
       .validateAt(name, { [name]: value })
       .then(() => {
         commitMutation({
           mutation: streamCollectionMutationFieldPatch,
           variables: {
-            id: props.streamCollection.id,
+            id: streamCollection.id,
             input: { key: name, value: value || '' },
           },
         });
       })
       .catch(() => false);
   };
+  const handleSubmitFieldOptions = (name, value) => streamCollectionValidation(t)
+    .validateAt(name, { [name]: value })
+    .then(() => {
+      commitMutation({
```

Comment:
I think you are missing error handling ?

Reply:
Not sure to understand this point, can you explain? Thanks

Reply:
After discussion with @Kedae its a more general problem that needs to be refactor at a higher level